src/Appwrite/Platform/Modules/Functions/Workers/Builds.php

```diff
@@ -833,56 +834,117 @@ protected function buildDeployment(Device $deviceForFunctions, Device $deviceFor
-                        ], $dbForPlatform, function (Document $rule) use ($dbForPlatform, $deployment) {
-                            $rule = $rule->setAttribute('value', $deployment->getId());
+                        $queries = [
+                            Query::equal("projectInternalId", [$project->getInternalId()]),
+                            Query::equal("type", ["deployment"]),
+                            Query::equal("deploymentResourceInternalId", [$resource->getInternalId()]),
+                            Query::equal('deploymentResourceType', ['site']),
+                            Query::equal('trigger', ['manual']),
+                        ];
+
+                        if (empty($oldDeploymentInternalId)) {
+                            $queries[] =  Query::equal("deploymentInternalId", [""]);
+                        } else {
+                            $queries[] = Query::equal("deploymentInternalId", [$oldDeploymentInternalId]);
+                        }
+
+                        $this->listRules($project, $queries, $dbForPlatform, function (Document $rule) use ($dbForPlatform, $deployment) {
+                            $rule = $rule
+                                ->setAttribute('deploymentId', $deployment->getId())
+                                ->setAttribute('deploymentInternalId', $deployment->getInternalId());
                             $dbForPlatform->updateDocument('rules', $rule->getId(), $rule);
                         });
 
-                        // VCS branch
-                        $branchName = $deployment->getAttribute('providerBranch');
-                        if (!empty($branchName)) {
-                            $this->listRules($project, [
-                                Query::equal("automation", ["branch=" . $branchName]),
-                            ], $dbForPlatform, function (Document $rule) use ($dbForPlatform, $deployment) {
-                                $rule = $rule->setAttribute('value', $deployment->getId());
-                                $dbForPlatform->updateDocument('rules', $rule->getId(), $rule);
-                            });
-                        }
-
-                        // VCS commit
-                        $commitHash = $deployment->getAttribute('providerCommitHash', '');
-                        if (!empty($commitHash)) {
-                            $this->listRules($project, [
-                                Query::equal("automation", ["commit=" . $commitHash]),
-                            ], $dbForPlatform, function (Document $rule) use ($dbForPlatform, $deployment) {
-                                $rule = $rule->setAttribute('value', $deployment->getId());
-                                $dbForPlatform->updateDocument('rules', $rule->getId(), $rule);
-                            });
-                        }
                         break;
                 }
             }
 
+            if ($resource->getCollection() === 'sites') {
```

Comment:
Same as in the other PR, this should be a build feature not a hardcoded check for sites.